backend/index.js

```diff
@@ -78,5 +72,43 @@ process.on('unhandledRejection', (reason, p) => {
   process.exit(1)
 })
 
+const exit = (code: number) => {
+  // Make sure `process.exit` is called after all existing SERVER_EXITING
+  // handlers. This is because once `process.exit` is called, all handlers
+  // must be synchronous.
+  sbp('okTurtles.events/once', SERVER_EXITING, () => {
+    // In case there are asynchronous events, wait for them to finish
+    sbp('okTurtles.eventQueue/queueEvent', SERVER_EXITING, () => {
+      process.send({}) // tell grunt we've successfully shutdown the server
+      process.nextTick(() => process.exit(code)) // triple-check we quit :P
+    })
+  })
+  sbp('okTurtles.events/emit', SERVER_EXITING)
+}
+
+const handleSignal = (signal: string, code: number) => {
+  process.on(signal, () => {
+    console.error(`Exiting upon receiving ${signal} (${code})`)
+    // Exit codes follow the 128 + signal code convention.
+    // See <https://tldp.org/LDP/abs/html/exitcodes.html>
+    exit(128 + code)
+  })
+}
+
+// Codes from <signal.h>
+;[
+  ['SIGHUP', 1],
+  ['SIGINT', 2],
+  ['SIGQUIT', 3],
+  ['SIGTERM', 15],
+  ['SIGUSR1', 10],
+  ['SIGUSR2', 11]
+].forEach(([signal, code]) => handleSignal(signal, code))
```

Comment:
So I'm trying to figure out under what circumstances the server shuts down with a `0` status code... and I don't see one except for receiving the `process.on('message'` below... this seems fine, but what about integration with the `chel serve` PR, will it work properly there too?